src/bridge/IrcHandler.ts

```diff
@@ -278,26 +290,24 @@ export class IrcHandler {
             if (!pmRoom) {
                 req.log.info("Creating a PM room with %s", bridgedIrcClient.userId);
                 this.pmRoomPromises[pmRoomPromiseId] = this.createPmRoom(
-                    bridgedIrcClient.userId, virtualMatrixUser.getId(), fromUser.nick, server
+                    req, bridgedIrcClient.userId, virtualMatrixUser.getId(), fromUser.nick, server
                 );
                 pmRoom = await this.pmRoomPromises[pmRoomPromiseId];
             }
         }
-        else {
-            // make sure that the matrix user is still in the room
-            try {
-                await this.ensureMatrixUserJoined(
-                    pmRoom.getId(), bridgedIrcClient.userId, virtualMatrixUser.getId(), req.log
-                );
-            }
-            catch (err) {
-                // We still want to send the message into the room even if we can't check -
-                // maybe the room state API has blown up.
-                req.log.error(
-                    "Failed to ensure matrix user %s was joined to the existing PM room %s : %s",
-                    bridgedIrcClient.userId, pmRoom.getId(), err
-                );
-            }
+        // make sure that the matrix user is (still) in the room
+        try {
+            await this.ensureMatrixUserJoined(
```

Comment:
Because I removed the invite on room creation, I currently have the issue that `ensureMatrixUserJoined` throws on the first message.

I lifted this out of its if block to invite the Matrix user right after the room creation.

```
2021-04-08 14:30:20 INFO:IrcBridge [-] POST http://localhost:8008/_matrix/client/r0/createRoom (@irc_lulu3:localhost) HTTP 200 "{\"room_id\":\"!ffNhtAuQcTJUElXPAg:localhost\"}"
2021-04-08 14:30:20 DEBUG:PgDatastore setPmRoom (matrix_user_id=@jaller94:localhost, virtual_user_id=@irc_lulu3:localhost, room_id=!ffNhtAuQcTJUElXPAg:localhost, irc_nick=lulu3)
2021-04-08 14:30:20 DEBUG:req [8g4thnt0c0k00] [[I->M]] Querying PM room state (!ffNhtAuQcTJUElXPAg:localhost) between @jaller94:localhost and @irc_lulu3:localhost
2021-04-08 14:30:20 INFO:IrcBridge [-] GET http://localhost:8008/_matrix/client/r0/rooms/!ffNhtAuQcTJUElXPAg%3Alocalhost/state/m.room.member/%40jaller94%3Alocalhost (@irc_lulu3:localhost) Body: 
2021-04-08 14:30:20 ERROR:IrcBridge [-] GET http://localhost:8008/_matrix/client/r0/rooms/!ffNhtAuQcTJUElXPAg%3Alocalhost/state/m.room.member/%40jaller94%3Alocalhost (@irc_lulu3:localhost) HTTP 404 Error: "{\"errcode\":\"M_NOT_FOUND\",\"error\":\"Event not found.\"}"
2021-04-08 14:30:20 ERROR:req [8g4thnt0c0k00] [[I->M]] Failed to ensure matrix user @jaller94:localhost was joined to the PM room !ffNhtAuQcTJUElXPAg:localhost : M_NOT_FOUND: Event not found.
    at parseErrorResponse (/home/jaller94/Git/matrix-appservice-irc/node_modules/matrix-js-sdk/lib/http-api.js:826:13)
    at /home/jaller94/Git/matrix-appservice-irc/node_modules/matrix-js-sdk/lib/http-api.js:781:17
    at /home/jaller94/Git/matrix-appservice-irc/node_modules/matrix-js-sdk/lib/http-api.js:715:9
    at Request._callback (/home/jaller94/Git/matrix-appservice-irc/node_modules/matrix-appservice-bridge/lib/components/client-factory.js:79:17)
    at Request.self.callback (/home/jaller94/Git/matrix-appservice-irc/node_modules/request/request.js:185:22)
    at Request.emit (node:events:378:20)
    at Request.EventEmitter.emit (node:domain:470:12)
    at Request.<anonymous> (/home/jaller94/Git/matrix-appservice-irc/node_modules/request/request.js:1154:10)
    at Request.emit (node:events:378:20)
    at Request.EventEmitter.emit (node:domain:470:12)
    at IncomingMessage.<anonymous> (/home/jaller94/Git/matrix-appservice-irc/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (node:events:484:28)
    at IncomingMessage.emit (node:events:390:22)
    at IncomingMessage.EventEmitter.emit (node:domain:470:12)
    at endReadableNT (node:internal/streams/readable:1307:12)
    at processTicksAndRejections (node:internal/process/task_queues:81:21) {
  errcode: 'M_NOT_FOUND',
  data: [Object],
  httpStatus: 404
}
2021-04-08 14:30:20 INFO:req [8g4thnt0c0k00] [[I->M]] Relaying PM in room !ffNhtAuQcTJUElXPAg:localhost
Apr-8 14:30:20.611 DEBUG js-sdk:matrix sendEvent of type m.room.message in !ffNhtAuQcTJUElXPAg:localhost with txnId m1617885020611.0
2021-04-08 14:30:20 INFO:IrcBridge [-] PUT http://localhost:8008/_matrix/client/r0/rooms/!ffNhtAuQcTJUElXPAg%3Alocalhost/send/m.room.message/m1617885020611.0 (@irc_lulu3:localhost) Body: "{\"msgtype\":\"m.text\",\"body\":\"Hi there\"}"
2021-04-08 14:30:20 INFO:IrcBridge 127.0.0.1 - - [08/Apr/2021:12:30:20 +0000] "PUT /transactions/919?access_token=<REDACTED> HTTP/1.1" 200 2 "-" "Synapse/1.29.0"

2021-04-08 14:30:20 INFO:IrcBridge [9x0iez3s64c00] [M->I] SUCCESS  (3ms)
2021-04-08 14:30:20 INFO:IrcBridge [-] PUT http://localhost:8008/_matrix/client/r0/rooms/!ffNhtAuQcTJUElXPAg%3Alocalhost/send/m.room.message/m1617885020611.0 (@irc_lulu3:localhost) HTTP 200 "{\"event_id\":\"$-VB6cXcbTPY0ovLY0XE9oBbt_GN5onI0pUvglhdUVfI\"}"
Apr-8 14:30:20.694 DEBUG js-sdk:matrix Event sent to !ffNhtAuQcTJUElXPAg:localhost with event id $-VB6cXcbTPY0ovLY0XE9oBbt_GN5onI0pUvglhdUVfI
2021-04-08 14:30:20 INFO:IrcBridge [8g4thnt0c0k00] [I->M] SUCCESS  (939ms)
```

Reply:
Looks a bit like it's failing to find the member state (expected), but it should be catching that and joining. Interesting.

Reply:
https://github.com/matrix-org/matrix-appservice-bridge/blob/6831acc9cfaf14ce67502ed8302e02422d2fee75/src/components/intent.ts#L746-L757

Reply:
Moved to a different PR (#1290) to not hold back the main change: Retrying the room creation.